test/e2e/util.go

```diff
@@ -1379,3 +1468,39 @@ func isPodScheduled(pod *v1.Pod) bool {
 	}
 	return false
 }
+
+func isJobOnlyOnePod(job *batchv1alpha1.Job) bool {
+	if len(job.Spec.Tasks) == 1 {
+		if job.Spec.Tasks[0].Replicas == 1 {
+			return true
+		}
+	}
+	return false
+}
+
+func getJobNode(ctx *testContext, job *batchv1alpha1.Job) (nodeName string, err error) {
+	if !isJobOnlyOnePod(job) {
+		return "", fmt.Errorf("job owns pods more than one")
+	}
+
+	pods, err := ctx.kubeclient.CoreV1().Pods(job.Namespace).List(context.TODO(), metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("volcano.sh/job-name=%s", job.Name),
+	})
+
+	if len(pods.Items) == 1 {
+		return pods.Items[0].Spec.NodeName, nil
+	}
+
+	return "", fmt.Errorf("get pods num not equal 1")
+}
+
+func compareNodename(ctx *testContext, jobA, jobB *batchv1alpha1.Job) (err error) {
```

Comment:
We can not guarantee the given job is of one replica.  May cause misuse by other developers